core/services/job_subscriber.go

```diff
@@ -37,16 +39,41 @@ type jobSubscriber struct {
 	jobSubscriptions map[string]JobSubscription
 	jobsMutex        *sync.RWMutex
 	runManager       RunManager
+	jobResumer       SleeperTask
+	runWorker        *runWorker
+}
+
+type runWorker struct {
+	runManager RunManager
+	head       big.Int
+}
```

Comment:
The PR in general looks good.  My only concern is maybe it could be done clearer.  For instance runWorker seems like a very general name.

Also, pro-tip, you can have function definitions satisfy interfaces, which may or may not help (like http.HandlerFunc)

Reply:
Needs to be a struct to hold the most recently observed head.

Reply:
Thats :+1:

My concern is for new comers to the code base.  What do you think about naming runWorker to something which describes what it does?  Also, the relation between jobResumer and runWorker as struct fields might be better captured in the field names no?  They directly relate to each other, but they appear unrelated.

For instance the line below looks like it involves two unrelated concepts:
```
js.runWorker.head = *head.ToInt()                                                                                                      js.jobResumer.WakeUp()        
```